### DISCUSSION TITLE
Filepaths containing whitespaces broken apart in all_changed_files

### DISCUSSION THREAD

**Discussion by rjwignar** [tj-actions/changed-files#1738] [Q&A] [answered]
### Background Information
Hi everyone,
I'm working on using **tj-actions/changed-files** to speed up the linting workflow of an Open Source project.
The idea is to use `tj-actions/changed-files` to pull a list of all modified files and pass that list to a linter step:
```yaml
# actions/checkout@v3 above
      - name: Get changed files
        id: changed-files
        uses: tj-actions/changed-files@v28.0.0
        with:
          files: |
            src/Features/*.cpp
            src/Features/*.h
            **/*.hlsl
            **/*.hlsli
      - name: List all changed files
        run: |
          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
            echo "$file was changed"
          done
# pass ${{ steps.changed-files.outputs.all_changed_files }} to linter
```

### Problem - Filepaths with whitespace not preserved
When testing this out, I noticed some of the files I modified contained whitespaces ` ` in their filepaths, which were broken up:

For example, after modifying a file with the filepath (note the spaces in `Complex Parallax Materials`):

```features/Complex Parallax Materials/Shaders/ComplexParallaxMaterials/ComplexParallaxMaterials.hlsli```

the 'List all changed files' step prints the following:

![image](https://github.com/tj-actions/changed-files/assets/78163326/58cca46d-8e2d-4e42-ae45-df4ede233109)

In short, the above filepath is broken into the following:
```
features/Complex
Parallax
Materials/Shaders/ComplexParallaxMaterials/CRPM.hlsli
```
which are then passed to the linter step and break CI.

### Question - How to preserve file paths containing whitespaces?
Is there a way to preserve the full pathnames?
Renaming the directories is out of the question, as I want my PR to only modify the workflow.

**Comment by rjwignar** [comment, marked as answer]
Currently attempting the solution proposed in https://github.com/tj-actions/changed-files/issues/609#issuecomment-1236559836:

> @preritdas Seems you'll need to tweak the code you have to resolve the issue.
> 
> For example, you'll need to use
> 
> ```yaml
> ...
>       - name: List all modified files
>          run: |
>            IFS=$',' read -a MODIFIED_FILES_ARRAY <<< "${{ steps.changed-files-comma.outputs.modified_files }}"
>            for file in "${MODIFIED_FILES_ARRAY[@]}"; do
>              echo $file
>            done
>            unset IFS
>          shell:
>            bash
> ```
> 
> See this [PR](https://github.com/tj-actions/changed-files/pull/611) for more information.
